src/libexpr/eval.cc

```diff
@@ -973,22 +973,20 @@ inline Value * EvalState::lookupVar(Env * env, const ExprVar & var, bool noEval)
 
     if (!var.fromWith) return env->values[var.displ];
 
+    if (noEval) return nullptr;
```

Comment:
It doesn't look like this can be moved here without consequence. What if it's already been forced? `noEval` is used by `maybeThunk`, so exiting too soon makes that optimization less effective.

Reply:
:+1:
Let's explain the trade-off.

```suggestion
    // This early exit defeats the `maybeThunk` optimization for variables from `with`,
    // The added complexity of handling this appears to be similarly in cost, or
    // the cases where applicable were insignificant in the first place.
    if (noEval) return nullptr;
```